### PR TITLE
Test follow-up reminders through real E2E flow with AI tracker creation

### PR DESCRIPTION
**Pull request by elie222**
# User description
## Summary

Convert follow-up reminders E2E tests from bypassing ThreadTracker creation to using real AI-powered flows. ThreadTrackers are now created by the actual conversation tracking system, making the tests true end-to-end integration tests.

## Changes

- Added `ensureConversationRules()` to enable conversation status rules needed for ThreadTracker creation
- Added `waitForThreadTracker()` polling helper to wait for AI to create trackers
- Added `disableNonConversationRules()` to prevent rule interference while keeping tracker creation enabled
- Rewrote 6 main tests to use real E2E flows (Gmail/Outlook AWAITING with/without draft, NEEDS_REPLY)
- Kept 3 edge case tests using direct DB insertion for efficiency

## Test Flow Changes

**Before**: Send email → directly insert ThreadTracker → test follow-up processing
**After**: Send email → reply/content triggers AI analysis → ThreadTracker created automatically → test follow-up processing

This ensures we're testing the full real integration path that users experience.

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
Enhances the follow-up reminder E2E tests by replacing manual database insertions with real AI-powered conversation tracking flows. Implements new polling helpers and rule management utilities to ensure tests accurately reflect the production user experience for Gmail and Outlook.
<table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1365?tool=ast&topic=Test+Infrastructure>Test Infrastructure</a>
        </td><td>Introduce <code>waitForThreadTracker</code> polling and rule management functions like <code>ensureConversationRules</code> to support automated AI flow verification.<details><summary>Modified files (2)</summary><ul><li>apps/web/__tests__/e2e/flows/helpers/accounts.ts</li>
<li>apps/web/__tests__/e2e/flows/helpers/polling.ts</li></ul></details><details><summary>Latest Contributors(1)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>Add-E2E-tests-for-foll...</td><td>January 20, 2026</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1365?tool=ast&topic=E2E+Flow+Refactor>E2E Flow Refactor</a>
        </td><td>Refactor follow-up reminder tests to use real email interactions and AI-triggered <code>ThreadTracker</code> creation instead of direct database mocks.<details><summary>Modified files (1)</summary><ul><li>apps/web/__tests__/e2e/flows/follow-up-reminders.test.ts</li></ul></details><details><summary>Latest Contributors(1)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>Isolate-follow-up-remi...</td><td>January 21, 2026</td></tr></table></details></td></tr></table>
This pull request is reviewed by Baz. Review like a pro on <a href=https://baz.co/changes/elie222/inbox-zero/1365?tool=ast>(Baz)</a>.

### CODE REVIEW

File: apps/web/__tests__/e2e/flows/follow-up-reminders.test.ts [hunk 1]
```diff
@@ -181,35 +182,51 @@ describe.skipIf(!shouldRunFlowTests())("Follow-up Reminders", () => {
...
         // ========================================
-        logStep("Step 3: Configuring follow-up settings");
+        logStep("Step 4: Configuring follow-up settings");
 
         await configureFollowUpSettings(gmail.id, {
-          followUpAwaitingReplyDays: 0.001, // ~1.4 minutes (less than 5 min ago)
+          followUpAwaitingReplyDays: 0.001, // ~1.4 minutes (threshold already passed)
           followUpNeedsReplyDays: null,
           followUpAutoDraftEnabled: true,
         });
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/__tests__/e2e/flows/follow-up-reminders.test.ts:229]: Good catch! Fixed in 6b73240 - changed all occurrences from `0.001` to `0` so newly-created ThreadTrackers pass the threshold check immediately.

[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/__tests__/e2e/flows/follow-up-reminders.test.ts:229]: Updated in e844295 - using `0.0001` days (~8.6 seconds) instead of `0`. This keeps realistic threshold behavior while the webhook + AI processing time (10-30s) reliably exceeds it.

[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/__tests__/e2e/flows/follow-up-reminders.test.ts:229]: Added explicit 10-second sleep after tracker creation in 7a2b3cb. This ensures tracker.sentAt is definitely past the 0.0001 days (~8.6s) threshold, making tests deterministic.
